lib/rewardable/Limiter.cpp

```diff
@@ -72,7 +72,7 @@ bool Rewardable::Limiter::heroAllowed(const CGHeroInstance * hero) const
 
 	if(daysPassed != 0)
 	{
-		if (IObjectInterface::cb->getDate(Date::DAY) < daysPassed)
+		if (IObjectInterface::cb->getDate(Date::DAY) <= daysPassed)
```

Comment:
Please recheck Water Wheel for possible off-by-one error.
It should give 500 gold on 1st week (days 1-7) and 1000 gold later (days 8+)

Reply:
and probably derricks and other rewardable from mods... maybe adding 1 during parsing hota map would be better solution